src/baseWrapper.ts

```diff
@@ -45,40 +43,28 @@ export default class BaseWrapper<ElementType extends Element> {
     return true
   }
 
-  async trigger(eventString: DomEventName | string, options?: TriggerOptions) {
-    if (options && options['target']) {
-      throw Error(
-        `[vue-test-utils]: you cannot set the target value of an event. See the notes section ` +
-          `of the docs for more details—` +
-          `https://vue-test-utils.vuejs.org/api/wrapper/trigger.html`
-      )
-    }
-
-    const isDisabled = () => {
-      const validTagsToBeDisabled = [
-        'BUTTON',
-        'COMMAND',
-        'FIELDSET',
-        'KEYGEN',
-        'OPTGROUP',
-        'OPTION',
-        'SELECT',
-        'TEXTAREA',
-        'INPUT'
-      ]
-      const hasDisabledAttribute = this.attributes().disabled !== undefined
-      const elementCanBeDisabled = validTagsToBeDisabled.includes(
-        this.element.tagName
-      )
-
-      return hasDisabledAttribute && elementCanBeDisabled
-    }
+  protected isDisabled = () => {
+    const validTagsToBeDisabled = [
+      'BUTTON',
+      'COMMAND',
+      'FIELDSET',
+      'KEYGEN',
+      'OPTGROUP',
+      'OPTION',
+      'SELECT',
+      'TEXTAREA',
+      'INPUT'
+    ]
+    const hasDisabledAttribute = this.attributes().disabled !== undefined
+    const elementCanBeDisabled = validTagsToBeDisabled.includes(
+      this.element.tagName
+    )
 
-    if (this.element && !isDisabled()) {
-      const event = createDOMEvent(eventString, options)
-      this.element.dispatchEvent(event)
-    }
-
-    return nextTick()
+    return hasDisabledAttribute && elementCanBeDisabled
   }
+
+  abstract trigger(
+    eventString: string,
+    options?: TriggerOptions
+  ): ReturnType<typeof nextTick>
 }
```

Comment:
Maybe I'm missing something, but the methods look very similar in both cases. Is it needed to have an abstract base method?

nit `typeof nextTick` can be written `Promise<void>` maybe?